alt-tab-macos/ui/Application.swift

```diff
@@ -93,4 +102,22 @@ class Application: NSApplication, NSApplicationDelegate, NSWindowDelegate {
         hideUi()
         DispatchQueue.global(qos: .userInteractive).async { window?.focus() }
     }
+
+    func closeSelectedWindow(_ window: TrackedWindow?) {
+        DispatchQueue.global(qos: .userInteractive).async { window?.close() }
+        isOutdated = true
+        // slight delay needed to give the window time to close before updating the thumbnails
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + .milliseconds(75), execute: {
+            self.showUiOrCycleSelection(TrackedWindows.focusedWindowIndex)
+        })
+    }
+
+    func quitApplicationOfSelectedWindow(_ window: TrackedWindow?) {
+        DispatchQueue.global(qos: .userInteractive).async { window?.quitApp() }
+        isOutdated = true
+        // slightly more delay needed to give the application time to quit before updating the thumbnails
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + .milliseconds(150), execute: {
+            self.showUiOrCycleSelection(TrackedWindows.focusedWindowIndex)
```

Comment:
Same comment as above for the window delay: delay won't cut it as some apps take multiple seconds and some apps take a few dozen ms.

Reply:
Same as in my [previous comment](https://github.com/lwouis/alt-tab-macos/pull/94#discussion_r362091513).